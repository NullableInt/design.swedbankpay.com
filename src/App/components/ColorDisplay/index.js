import React from "react";
import PropTypes from "prop-types";
import ColorPreview from "@components/ColorPreview";

const ColorDisplay = ({ id, colorList, head, description, subListNames }) => (
    <>
        <h2 id={id}>{head}</h2>
        {description}

        {subListNames ? subListNames.map((subListName, index) => (
            <div key={index}>
                <h3>{subListName}</h3>
                <ul className="list color-preview">
                    {colorList.filter(elem => elem.inSubLists.includes(subListName)).map((elem, i) => (
                        <li key={i} >
                            <ColorPreview name={elem.name} hex={elem.hex} />
                        </li>
                    ))}
                </ul>
            </div>))
            :
            <ul className="list color-preview">
                {colorList.map((elem, i) => (
                    <li key={i} >
                        <ColorPreview name={elem.name} hex={elem.hex} />
                    </li>
                ))}
            </ul>
        }
    </>
);

ColorDisplay.propTypes = {
    id: PropTypes.string.isRequired,
    colorList: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        hex: PropTypes.string.isRequired,
        subListNames: PropTypes.arrayOf(PropTypes.string)
    })).isRequired,
    head: PropTypes.string.isRequired,
    description: PropTypes.any,
    subLists: PropTypes.bool
};

export default ColorDisplay;
