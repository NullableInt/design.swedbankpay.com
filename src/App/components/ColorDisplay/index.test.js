import React from "react";
import { shallow } from "enzyme";

import ColorDisplay from "./index";

describe("Component: ColorDisplay", () => {
    it("is defined", () => {
        expect(ColorDisplay).toBeDefined();
    });

    it("ColorDisplay renders", () => {
        const mockColor = {
            name: "Yellow",
            hex: "#FDC129"
        };

        const mockColorDisplayElement = {
            id: "test",
            colorList: [mockColor, mockColor],
            head: "Test Head",
            description: null
        };

        const wrapper = shallow(<ColorDisplay {...mockColorDisplayElement}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
