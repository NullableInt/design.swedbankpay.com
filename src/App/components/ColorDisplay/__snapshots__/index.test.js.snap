// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ColorDisplay ColorDisplay renders 1`] = `
<Fragment>
  <h2
    id="test"
  >
    Test Head
  </h2>
  <ul
    className="list color-preview"
  >
    <li
      key="0"
    >
      <ColorPreview
        hex="#FDC129"
        name="Yellow"
      />
    </li>
    <li
      key="1"
    >
      <ColorPreview
        hex="#FDC129"
        name="Yellow"
      />
    </li>
  </ul>
</Fragment>
`;
